server/src/Handler/Item.hs

```diff
@@ -79,17 +85,21 @@ validateItemPrice price =
         then Left "Price should be above 0"
         else Right price
 
-itemForm :: UserId -> Maybe Item -> Form Item
+itemForm :: UserId -> Maybe Item -> Form (Item, (FileInfo, UTCTime))
 itemForm userId mitem =
     renderSematnicUiDivs $
-    Item <$> areq textField "Name" (itemName <$> mitem) <*>
-    areq
-        (selectField companies)
-        (selectSettings "Company")
-        (itemCompany <$> mitem) <*>
-    areq priceField "Price" (itemPrice <$> mitem) <*>
-    lift (liftIO getCurrentTime) <*>
-    pure userId
+    (,) <$>
+    (Item <$> areq textField "Name" (itemName <$> mitem) <*>
+     areq
+         (selectField companies)
+         (selectSettings "Company")
+         (itemCompany <$> mitem) <*>
+     areq priceField "Price" (itemPrice <$> mitem) <*>
+     -- @todo: Why doesn't `map itemPdfFile mitem` work here?
+     pure (maybe Nothing itemPdfFile mitem) <*>
```

Comment:
@zarvunk hi! quick question: `mitem` is a Maybe value, and `itemPdfFile` is a name of one of it's record properties. when I try to `fmap itemPdfFile mitem` I get:

```
    • Couldn't match type ‘Maybe (Key PdfFile)’ with ‘Key PdfFile’
      Expected type: AForm (HandlerT App IO) (Maybe (Key PdfFile))
        Actual type: AForm
                       (HandlerT App IO) (Maybe (Maybe (Key PdfFile)))
```

I was assuming the fmap should return just a single `Maybe`, same as when I do:

![client_ _ghci_ _ghc_ _119x30](https://user-images.githubusercontent.com/125707/27347348-f8a7776a-55f8-11e7-8695-d24cccf59ef0.jpg)


What am I'm missing? :)